### PR TITLE
boards: add support for nRF52840dk

### PR DESCRIPTION
**Pull request by haukepetersen**
Adding support for the new `nRF52840 Preview DK` by Nordic. Although the current HW is labeled as preview DK, IMHO it makes sense not to include the preview in the name, as by experience the differences from the 'preview' version to the final version are transparent to RIOT.

### CODE REVIEW

File: cpu/nrf51/periph/i2c.c [hunk 1]
```diff
@@ -101,8 +101,8 @@ int i2c_init_master(i2c_t bus, i2c_speed_t speed)
     dev(bus)->POWER = TWI_POWER_POWER_Enabled;
 
     /* pin configuration */
-    GPIO_BASE->PIN_CNF[i2c_config[bus].pin_scl] = (GPIO_PIN_CNF_DRIVE_S0D1 << GPIO_PIN_CNF_DRIVE_Pos);
-    GPIO_BASE->PIN_CNF[i2c_config[bus].pin_scl] = (GPIO_PIN_CNF_DRIVE_S0D1 << GPIO_PIN_CNF_DRIVE_Pos);
+    NRF_GPIO->PIN_CNF[i2c_config[bus].pin_scl] = (GPIO_PIN_CNF_DRIVE_S0D1 << GPIO_PIN_CNF_DRIVE_Pos);
```
Comment (aabadie) [inline cpu/nrf51/periph/i2c.c]: The changes here are not related to the commit comment. I think they should be in a separated one or dropped.

Comment (haukepetersen) [inline cpu/nrf51/periph/i2c.c]: Yes, they are related, as `GPIO_BASE` does not exist anymore... Adapted the commit message.

**Comment by haukepetersen** [comment]
fixed dos line endings and newlines at EOF for vendor headers

**Comment by haukepetersen** [comment]
removed softdevice support (for now) and fixed one typo in the uart driver that Murdock found.
